A. Did some corrections to docstring of sylvester, and some changes/a…

akritas opened the pull request:
…dditions

to the general docstring.

B. Added the function res(f, g, x) to compute the resultant of f, g
by evaluating the determinant of sylvester(f, g, x, 1).

Added this function because all cas's return 0 for resultant(f, 0),
which imho is wrong because f and 0 do _not_ have a common divisor.
The new function returns a statement that the resultant is _not_ defined.

akritas committed:
A. Did some corrections to docstring of sylvester, and some changes/additions
to the general docstring.

B. Added the function res(f, g, x) to compute the resultant of f, g
by evaluating the determinant of sylvester(f, g, x, 1).

Added this function because all cas's return 0 for resultant(f, 0),
which imho is wrong because f and 0 do _not_ have a common divisor.
The new function returns a statement that the resultant is _not_ defined.

asmeurer commented on sympy/polys/subresultants_qq_zz.py:
```diff
@@ -326,6 +329,20 @@ def sylvester(f, g, x, method = 1):
+
+    """
+    if f == 0 or g == 0:
+        print("The resultant of ",f," and ", g," is not defined")
```
Should raise an exception here (like `ValueError("The resultant of %s and %s is not defined" % (f, g))`).

asmeurer replied:
Actually, instead of ValueError it should probably be PolynomialError (or one of the more specific errors from sympy/polys/polyerrors.py).

jksuom commented:
The following is usually accepted as a standard property of the resultant of two polynomials over a field K.

The resultant is zero if and only if the polynomials have a common root in some extension of K.

When one of the polynomials is zero this means that the resultant is non-zero if and only if the other polynomial  has no root in any extension, or, equivalently, if and only if it is a non-zero constant.
Hence it seems that ``res(f, 0)`` should be zero unless f is such a constant.

asmeurer commented:
The resultant function already in SymPy gives

```
In [3]: resultant(0, 1, x)
Out[3]: 0

In [4]: resultant(0, 0, x)
Out[4]: 0

In [5]: resultant(0, x)
Out[5]: 0
```

jksuom commented:
The ``resultant(0, 1, x)`` case should probably be fixed, in principle.

akritas commented:
Yes, but  looking at Joel Cohen’s
book “Computer Algebra and Symbolic Computation” he says that when one  
of the arguments is 0
the resultant is not defined.

Actually, what the sympy resultant does is to take the last member of  
subresultants. BUT, in this case the prs consists of only the input  
"polys".

subresultants(0, 1, x) = subresultants(1, 0, x) = [1, 0];

so resultant just returns the 0 that was given as input.

Should I go ahead and change the print to an error statement or drop  
this function and reference to it altogether?  I guess, if you do not  
want me to include res() in the module, I can always use it separately  
on my own.

Quoting Aaron Meurer <notifications@github.com>:

> The resultant function already in SymPy gives
>
> ```
> In [3]: resultant(0, 1, x)
> Out[3]: 0
>
> In [4]: resultant(0, 0, x)
> Out[4]: 0
>
> In [5]: resultant(0, x)
> Out[5]: 0
> ```
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/sympy/sympy/pull/10694#issuecomment-189460491

akritas committed:
replaced print with ValueError (PolynomialError gave me problems)
in res(f, g, x); also gave a reference for a good definition of res()

made a test for res()

akritas committed:
changed ValueError to PolynomialError

akritas committed:
1. Added some more comments in the docstring of the module,
2. added res_q() and res_z(), two more functions for computing
  the resultant of two polys,
3. added tests

akritas committed:
Added subresultants_sylv() and modified_subresultants_sylv()
to compute by determinant evaluation the prs's.

Also changed the name of the function process_bezout_output to
process_matrix_output, since it is also needed by the two new
functions.

Added tests for the two new functions
